<!--
    Sets the parameters and brings up the l_cart and r_cart controllers.
    They're brought up in a stopped state; once the robot initial pose is set, they'll be activated.
-->

<launch>

  <rosparam ns="r_cart">
    type: teleop_controllers/JinvTeleopController7
    root_name: torso_lift_link
    tip_name: r_gripper_tool_frame
    k_posture: 20.0
    jacobian_inverse_damping: 0.01
    pose_command_filter: 0.01
    cart_gains:
      trans:
        p: 800.0
        d: 15.0
      rot:
        p: 80.0
        d: 1.2
    joints:
      r_shoulder_pan_joint:   {p: 2400.0, d: 18.0, ff: 10.0, p_clamp: 0.087}
      r_shoulder_lift_joint:  {p: 1200.0, d: 10.0, ff: 10.0, p_clamp: 0.087}
      r_upper_arm_roll_joint: {p: 1000.0, d: 6.0,  ff: 8.0,  p_clamp: 0.087}
      r_elbow_flex_joint:     {p: 700.0,  d: 4.0,  ff: 5.0,  p_clamp: 0.087}
      r_forearm_roll_joint:   {p: 300.0,  d: 6.0,  ff: 5.0,  p_clamp: 0.087}
      r_wrist_flex_joint:     {p: 300.0,  d: 4.0,  ff: 2.0,  p_clamp: 0.087}
      r_wrist_roll_joint:     {p: 300.0,  d: 4.0,  ff: 2.0,  p_clamp: 0.087}
    vel_saturation_trans: 1.0
    vel_saturation_rot: -1
  </rosparam>
  <rosparam ns="l_cart">
    type: teleop_controllers/JinvTeleopController7
    root_name: torso_lift_link
    tip_name: l_gripper_tool_frame
    k_posture: 20.0
    jacobian_inverse_damping: 0.01
    pose_command_filter: 0.01
    cart_gains:
      trans:
        p: 800.0
        d: 15.0
      rot:
        p: 80.0
        d: 1.2
    joints:
      l_shoulder_pan_joint:   {p: 2400.0, d: 18.0, ff: 10.0, p_clamp: 0.087}
      l_shoulder_lift_joint:  {p: 1200.0, d: 10.0, ff: 10.0, p_clamp: 0.087}
      l_upper_arm_roll_joint: {p: 1000.0, d: 6.0,  ff: 8.0,  p_clamp: 0.087}
      l_elbow_flex_joint:     {p: 700.0,  d: 4.0,  ff: 5.0,  p_clamp: 0.087}
      l_forearm_roll_joint:   {p: 300.0,  d: 6.0,  ff: 5.0,  p_clamp: 0.087}
      l_wrist_flex_joint:     {p: 300.0,  d: 4.0,  ff: 2.0,  p_clamp: 0.087}
      l_wrist_roll_joint:     {p: 300.0,  d: 4.0,  ff: 2.0,  p_clamp: 0.087}
    vel_saturation_trans: 1.0
    vel_saturation_rot: -1
  </rosparam>
  <node name="spawn_cart"
        pkg="pr2_controller_manager" type="spawner"
        args="--stopped r_cart l_cart" />
  

  <!-- Node that will switch to Cartesian controllers once the initial pose has been set -->
  <node name="switch_to_cartesian_controllers" output="screen"
        pkg="articulate_cart" type="switch_controllers.py" />
        
  
</launch>
